app/controllers/course/courses_controller.rb

```diff
@@ -8,7 +8,7 @@ def index # :nodoc:
 
   def show # :nodoc:
     @registration = Course::Registration.new
-    @activity_feeds = recent_activity_feeds.limit(20).includes(:activity)
+    @activity_feeds = recent_activity_feeds.limit(20).includes(activity: [:object, :actor])
```

Comment:
Although the scope might be the best option, the issue with this implementation is that if the base object for the last 20 activities has been deleted, there wouldn't be an activity feed. That said, we don't expect that to happen a lot, so I guess this is a fair tradeoff.